src/figuresClasses.ts

```diff
@@ -1,11 +1,79 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - You don't need to manually assign constructor parameters to class properties if you use the 'public' keyword, as TypeScript does this automatically. You can remove the assignments to 'this.color', 'this.a', 'this.b', and 'this.c'.